cobigen-maven/cobigen-maven-plugin/src/main/java/com/devonfw/cobigen/maven/GenerateMojo.java

```diff
@@ -223,18 +222,26 @@ private CobiGen createCobiGenInstance() throws MojoExecutionException, MojoFailu
             }
         }
 
-        URL contextXmlUrl = classRealm.getResource("context.xml");
-        if (contextXmlUrl == null) {
-            getLog().error("No context.xml could be found in the classpath!");
-            return null;
+        Path rootPath;
+        URL rootUrl = classRealm.getResource("context.xml");
+        if (rootUrl == null) {
+            rootUrl = classRealm.getResource("src/main/templates/context.xml");
+            if (rootUrl == null) {
+                getLog().error("No context.xml could be found in the classpath!");
+                return null;
```

Comment:
Better throw a MojoFailureException and abort processing as not finding the context.xml would result in strange behavior as one of our main assumptions seems to be not valid anymore.